dashboard/dashboard.py

```diff
@@ -242,19 +252,44 @@ def analyze_experiment(info, experiments_dir, tmp_data_dir,
 
     status = validate_status(tmp_analysis_dir)
 
+    # collect data to dump to data_*.json
+    dump_data = {}
     # read the analyzed data, append a timestamp field, and copy over to the permanent data dir
     if status['success']:
         data_exists = check_file_exists(tmp_analysis_dir, 'data.json')
         if not data_exists:
             status = {'success': False, 'message': 'No data.json file produced by {}'.format(exp_name)}
         else:
             data = read_json(tmp_analysis_dir, 'data.json')
-            data['timestamp'] = date_str
-            data['tvm_hash'] = tvm_hash
-            write_json(analyzed_data_dir, 'data_{}.json'.format(date_str), data)
-
+            data.update({
+                'timestamp'  : date_str,
+                'tvm_hash'   : tvm_hash,
+            })
+            dump_data.update(data)
+
+    # fetch time spent on the experiment
+    run_status = validate_json(info.exp_status_dir(exp_name), 
+                            'start_time', 
+                            'end_time', 
+                            'time_delta', filename='run.json')
+    # only process valid files
+    keys = run_status.keys()
+    if len(keys) and functools.reduce(lambda x, y: x and y, 
```

Comment:
Nitpick: I don't think it's actually necessary to test `if len(keys)`; `if keys` should be equivalent (and in this case, the reduce will turn out false if keys is empty)